### PR TITLE
Samples - show all/show assigned [SCI-1277]

### PR DESCRIPTION
**Pull request by rekonder**
https://biosistemika.atlassian.net/browse/SCI-1277

### CODE REVIEW

**Comment by mz3944** [comment]
We should wait for my [PR#643](https://github.com/biosistemika/scinote-web/pull/643) to be merged, and then you can rebase it on top of it, because otherwise I get error when switching to "View all samples".

**Comment by mz3944** [comment]
It's been merged, so you can rebase now.

**Comment by rekonder** [comment]
rebased!

File: app/datatables/sample_datatable.rb [hunk 1]
```diff
@@ -207,8 +218,12 @@ def get_raw_records
                           (sample_my_modules.my_module_id IN (#{ids.to_sql}) OR
                           sample_my_modules.id IS NULL))")
                           .references(:sample_my_modules)
+      if params[:assigned].present? && params[:assigned] == 'assigned'
+        samples = samples.joins('LEFT OUTER JOIN "my_modules" ON "my_modules"."id" = "sample_my_modules"."my_module_id"')
+                         .where('"my_modules"."experiment_id" = ?', @experiment.id)
+                         .where('"my_modules"."nr_of_assigned_samples" > 0')
+      end
     end
-
```
Comment (mz3944) [inline app/datatables/sample_datatable.rb]: I think you can move all three parts of the code here, as in: 

```
if params[:assigned] == 'assigned'
  samples = samples.where('"sample_my_modules"."id" > 0')
end
```
Correct me if I'm wrong...

**Comment by mz3944** [comment]
Merge away! :speedboat:

**Pull request merged**